src/dag.rs

```diff
@@ -125,4 +125,240 @@ where
 }
 
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::*;
+    use petgraph::graph::DiGraph;
+    use std::collections::HashSet;
+
+    #[test]
+    fn test_build_dag_success() {
+        // Test: Can we build a dag?
```

Comment:
@coderabbitai Please change all doc strings to be doc strings. This applies to all these in the PR.

Reply:
Lets not change all comments, just add simple doc stings to the unit test in `dag.rs`. Please follow the pattern in `db.rs` for `dag.rs`.